### PR TITLE
Closes #2112 add `type` to functions downstream of compute_duration()

### LINKED ISSUE #2112: Bug: `type` argument not available in `derive_vars_duration()`
### What happened?

In the last release we added the `type` argument to `compute_duration()` and changed the default behavior but we did not add it to `derive_vars_duration()`. I.e., the behavior of `derive_vars_duration()` has changed but there is no way for the users to get back the old behavior.

Thus we should add the `type` argument to `derive_vars_duration()`. Maybe we should also add it to `derive_vars_aage()`. The behavior of `derive_vars_aage()` was changed as well (most likely unintentional). At least the old behavior should be restored.

It should also checked if other functions might be affected by the change in `compute_duration()`.

### Session Information

_No response_

### Reproducible Example

_No response_

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Could we add some tests for this issue?

File: R/derive_vars_aage.R [hunk 1]
```diff
@@ -67,7 +91,8 @@ derive_vars_aage <- function(dataset,
                              start_date = BRTHDT,
                              end_date = RANDDT,
                              unit = "years",
-                             age_unit = "years") {
+                             age_unit = "years",
+                             type = "duration") {
```
Comment (bundfussr) [inline R/derive_vars_aage.R:95]: I would default `type` to `"interval"` because this is more common for age.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# admiral 0.12.1
+
+- Fix bug in `derive_vars_duration()` and `derive_vars_aage()` that adds `type` argument that gets passed through the `compute_duration()`. (#2112)
```
Comment (bundfussr) [inline NEWS.md:3]: Could you provide more details (what was changed in 0.12.0, what went wrong, what was changed in 0.12.1)?

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: R/derive_vars_aage.R [hunk 3]
```diff
@@ -37,12 +37,36 @@
...
+#'   Default: `"duration"`
+#'
+#'   Permitted Values: `"duration"`, `"interval"`
+#'
+#' @details The duration is derived as time from start to end date in the
+#'   specified output unit. If the end date is before the start date, the duration
+#'   is negative. The start and end date variable must be present in the specified
+#'   input dataset.
+#'
+#'   The [lubridate](https://lubridate.tidyverse.org/) package calculates two
```
Comment (bms63) [inline R/derive_vars_aage.R:53]: Is there a way for us inherit this from the source? derive_vars_duration??  Just so if we update it in the other function it populates here.

File: R/derive_vars_duration.R [hunk 4]
```diff
@@ -72,11 +72,37 @@
 #'
 #'   Permitted Values: `TRUE`, `FALSE`
 #'
+#' @param type lubridate duration type.
+#'
+#'   See below for details.
+#'
+#'   Default: `"duration"`
```
Comment (bms63) [inline R/derive_vars_duration.R:79]: Sorry - I'm really struggling to remember - Should we be purging Defaults from our documentation just so it is one less thing to maintain??  @bundfussr

Comment (bundfussr) [reply R/derive_vars_duration.R:79]: Yes, I think we decided not stating the default value in the parameter description.

File: R/derive_vars_aage.R (hunk 3 above)
Comment (bundfussr) [reply R/derive_vars_aage.R:53]: I think we should add a `roxygen_*()` function for this (once we have introduced this concept). As far as I know inheritance is only possible for `@param` tags.

**Review by bms63 — APPROVED** [review]
Think we are good to go

**Pull request merged**